File: Mario.c
#include <cs50.h>
#include <stdio.h>

void print_space(int n);
void print_hash(int n);
void draw_pyramid(int n);

int main(void)
{
    // Ask the user for height between 1 - 8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    
    // Loop if the input is not within range of 1-8, inclusive
    while (height < 1 || height > 8);
    draw_pyramid(height);   
}

// Print n number of spaces
void print_space(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf(" ");
    }
}

// Print n number of hashes
void print_hash(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("#");        
    }
}

// Draw the rows of the pyramid
void draw_pyramid(int n)
{
    // Where n is the height, will print number of rows equal to the height
    for (int i = 1; i <= n; i++)
    {
        print_space(n - i);
        print_hash(i);
        print_space(2);
        print_hash(i);
        printf("\n");
    }   
}

File: mario/more/mario.py
from cs50 import get_int


def main():
    # Ask the user for height between 1 - 8
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break

    draw_pyramid(height)


# Print n number of spaces
def print_space(n):
    for i in range(n):
        print(" ", end="")

# Print n number of hashes


def print_hash(n):
    for i in range(n):
        print("#", end="")


# Draw the rows of the pyramid
def draw_pyramid(n):
    # Where n is the height, will print number of rows equal to the height
    for i in range(1, (n+1)):
        print_space(n - i)
        print_hash(i)
        print_space(2)
        print_hash(i)
        print("")


if __name__ == "__main__":
    main()